sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRebaseDatetimeSuite.scala

```diff
@@ -164,7 +165,13 @@ abstract class ParquetRebaseDatetimeSuite
     def successInRead(path: String): Unit = spark.read.parquet(path).collect()
     def failInRead(path: String): Unit = {
       val e = intercept[SparkException](spark.read.parquet(path).collect())
-      assert(e.getCause.isInstanceOf[SparkUpgradeException])
+      // If SparkUpgradeException is thrown while reading file,
+      // it will be wrapped by QueryExecutionException.
+      if (e.getCause.isInstanceOf[QueryExecutionException]) {
```

Comment:
Do we want the same match pattern everywhere else, like here?

Reply:
```
e.getCause match {
  case QueryExecutionException => assert(...)
  case SparkUpgradeException => // OK
  case _ => fail(...)
}
```
Yeah maybe not cleaner, it's OK

Reply:
BTW, where do we wrap `SparkUpgradeException`? In `FileScanRDD` we intentionally avoid wrapping `SparkUpgradeException`.

Reply:
> BTW, where do we wrap `SparkUpgradeException`? In `FileScanRDD` we intentionally avoid wrapping `SparkUpgradeException`.

Seems origin code
```
case e: ParquetDecodingException =>
        if (e.getCause.isInstanceOf[SparkUpgradeException]) {
          throw e.getCause
        } else if (e.getMessage.contains("Can not read value at")) {
          throw QueryExecutionErrors.cannotReadParquetFilesError(e)
        }
        throw e
```
can't catch SparkUpgradeException, SparkUpgradeException is directly thrown, not wrapped by ParquetDecodingException

Reply:
> BTW, where do we wrap `SparkUpgradeException`? In `FileScanRDD` we intentionally avoid wrapping `SparkUpgradeException`.

Handle this,

Reply:
> ```
> e.getCause match {
>   case QueryExecutionException => assert(...)
>   case SparkUpgradeException => // OK
>   case _ => fail(...)
> }
> ```
> 
> Yeah maybe not cleaner, it's OK

Won't need this change now.